C Library can't build with just classic netCDF

When built with classic netCDF only (no netcdf-4, no pnetcdf), the build fails:

cd /var/lib/jenkins/workspace/PIO_develop_netcdf-4.4.1_classic_c_only/examples/c && /usr/bin/cmake -E cmake_link_script CMakeFiles/example1.dir/link.txt --verbose=1
/usr/local/bin/mpicc  -g -I/usr/local/netcdf-4.4.1-classic/include  -std=c99 -g -O0   CMakeFiles/example1.dir/example1.c.o  -o example1 -rdynamic ../../src/clib/libpioc.a /usr/local/netcdf-4.4.1-classic/lib/libnetcdf.so -Wl,-rpath,/usr/local/netcdf-4.4.1-classic/lib 
../../src/clib/libpioc.a(pio_nc.c.o): In function `PIOc_inq_var_fill':
/var/lib/jenkins/workspace/PIO_develop_netcdf-4.4.1_classic_c_only/src/clib/pio_nc.c:1952: undefined reference to `nc_inq_var_fill'
collect2: error: ld returned 1 exit status
examples/c/CMakeFiles/example1.dir/build.make:96: recipe for target 'examples/c/example1' failed
make[2]: *** [examples/c/example1] Error 1
make[2]: Leaving directory '/var/lib/jenkins/workspace/PIO_develop_netcdf-4.4.1_classic_c_only'
CMakeFiles/Makefile2:1715: recipe for target 'examples/c/CMakeFiles/example1.dir/all' failed
make[1]: *** [examples/c/CMakeFiles/example1.dir/all] Error 2
make[1]: Leaving directory '/var/lib/jenkins/workspace/PIO_develop_netcdf-4.4.1_classic_c_only'
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
Build step 'Execute shell' marked build as failure
Finished: FAILURE

This is the next build I have to get working in order to test the error handling properly. (That is, make sure the error handling works without netCDF-4 in the build).

--- a/src/clib/pio_nc.c
+++ b/src/clib/pio_nc.c
@@ -1947,9 +1947,12 @@ int PIOc_inq_var_fill(int ncid, int varid, int *no_fill, void *fill_valuep)
         if (file->iotype == PIO_IOTYPE_PNETCDF)
             ierr = ncmpi_inq_var_fill(file->fh, varid, no_fill, fill_valuep);
 #endif /* _PNETCDF */
-#ifdef _NETCDF
+#ifdef _NETCDF4
+        /* The inq_var_fill is not supported in classic-only builds. */
         if (file->iotype != PIO_IOTYPE_PNETCDF && file->do_io)
             ierr = nc_inq_var_fill(file->fh, varid, no_fill, fill_valuep);
+#else
+        ierr = PIO_ENOTNC4;
 #endif /* _NETCDF */
     }
 

